game_app/games/tic_tac_toe/tic_tac_toe_component.py

```diff
@@ -220,16 +216,36 @@ def step(self, position):
         self.turn = TurnState.NOT_YOUR_TURN
         row, col = position
         action = TicTacToeAction(row, col)
-        self.asys.tell(self.client_actor_address, MoveMsg(action))
+        self.tell(self.client_actor_address, MoveMsg(action))
 
     def restart(self):
+        self.turn = TurnState.NOT_YOUR_TURN
+        self.winnings = []
+        self._scene = TicTacToeScene(self, self._app.screen, self._board_size)
         self.server.restart(blocking=False)
-        self.log("called: self.server.restart(blocking=False)")
+        self.log("Restarted server")
 
     def back_to_menu(self):
         self.server.shutdown()
         self._app.switch_component(Components.MAIN_MENU)
 
-    def log(self, text):
+    def log(self, text, logging_level=LoggingLevel.GAME_EVENTS):
+        if not LOGGING:
+            return
         if self.logger_addr is not None:
-            self.asys.tell(self.logger_addr, LogMsg(text, "TicTacToeComponent"))
+            self.asys.tell(self.logger_addr, LogMsg(text, "TicTacToeComponent", logging_level))
+
+    def tell(self, target_address, message):
+        self.asys.tell(target_address, message)
+        if not isinstance(message, GetEventsToPostMsg):
+            self.log(f"Sent {message} to {target_address}", LoggingLevel.PLATFORM_COMMUNICATION_MESSAGES)
+
+    def listen(self):
+        response = self.asys.listen()
+        if not isinstance(response, EventsToPostMsg):
+            self.log(f"Received {response}", LoggingLevel.PLATFORM_COMMUNICATION_MESSAGES)
+        return response
+
+    def ask(self, target_address, message):
+        self.tell(target_address, message)
+        return self.listen()
```

Comment:
po co ta funkcja?
jak uwazasz ze robi czystosc to i tak zaimplementowalbym ja jako '''return self.asys.ask(target_address, message)'''

Reply:
jest używana w loopie, a gdyby był zaimplementowana tak jak piszesz to nie logowałaby wysłanych i odebranych wiadomości